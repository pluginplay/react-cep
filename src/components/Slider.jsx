import React from 'react'
import RCSlider from 'rc-slider/lib/Slider'
import 'rc-slider/assets/index.css'
import styled from 'styled-components'
import omit from 'lodash/omit'

class Slider extends React.Component {
  render () {
    return (
      <div className={this.props.className}>
        <RCSlider {...omit(this.props, ['className'])} />
      </div>
    )
  }
}

export default styled(Slider)`
  flex-grow: 1;
  margin: 0 15px;
  .rc-slider {
    .rc-slider-rail {
      background-color: ${props => props.theme.background.dark};
      border-radius: 0;
    }
    .rc-slider-track {
      background-color: ${props => props.theme.ae.systemHighlightColor};
      border-radius: 0;
    }
    .rc-slider-handle {
      border: none;
      width: 12px;
      height: 12px;
      margin-left: -6px;
      margin-top: -4px;
      background-color: ${props => props.theme.ae.systemHighlightColor};
    }
  }
`
