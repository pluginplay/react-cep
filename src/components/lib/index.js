import SystemState from '../../models/SystemState'
import PersistedState from '../../models/PersistedState'

export { SystemState, PersistedState }
