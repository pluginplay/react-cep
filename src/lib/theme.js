import * as csInterface from 'cep-lib/csinterface'
import { lighten, darken } from 'polished'

export const getDefaultTheme = () => {
  const ae = csInterface.getThemeInformation()
  ae.systemHighlightColor = '#0078d7' // We have to hard-code this because Windows gets the wrong color.
  return {
    ae,
    background: {
      lighter: lighten(0.2, ae.panelBackgroundColor),
      light: lighten(0.1, ae.panelBackgroundColor),
      dark: darken(0.05, ae.panelBackgroundColor),
      darker: darken(0.1, ae.panelBackgroundColor)
    },
    colors: {
      green: '#23EB87',
      red: '#eb093f',
      blue: ae.systemHighlightColor
    }
  }
}
