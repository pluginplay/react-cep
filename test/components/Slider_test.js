import { expect, withTheme } from '../setup'
import React from 'react'
import Slider from '../../src/components/Slider'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import { getDefaultTheme } from '../../src/lib/theme'

import RCSlider from 'rc-slider/lib/Slider'

it('renders correctly', () => {
  const tree = renderer
    .create(withTheme(<Slider min={0} max={100} value={10} />))
    .toJSON()
  jestExpect(tree).toMatchSnapshot()
})

it('forwards props', () => {
  const wrapper = shallow(<Slider theme={getDefaultTheme()} min={0} max={100} value={10} />).dive()
  const slider = wrapper.find(RCSlider).first()
  expect(slider).to.have.prop('min').equal(0)
  expect(slider).to.have.prop('max').equal(100)
  expect(slider).to.have.prop('value').equal(10)
  expect(slider).to.have.prop('className').equal('') // RCSlider attaches their own className by default.
})
