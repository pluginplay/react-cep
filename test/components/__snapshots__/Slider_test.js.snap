// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-bdVaJa cxWwFF"
>
  <div
    className="rc-slider"
    onBlur={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseDown={[Function]}
    onTouchStart={[Function]}
    style={undefined}
  >
    <div
      className="rc-slider-rail"
      style={Object {}}
    />
    <div
      className="rc-slider-track"
      style={
        Object {
          "left": "0%",
          "width": "10%",
        }
      }
    />
    <div
      className="rc-slider-step"
    />
    <div
      aria-disabled={false}
      aria-valuemax={100}
      aria-valuemin={0}
      aria-valuenow={10}
      className="rc-slider-handle"
      role="slider"
      style={
        Object {
          "left": "10%",
        }
      }
      tabIndex={0}
    />
    <div
      className="rc-slider-mark"
    />
  </div>
</div>
`;
